### PR TITLE
Use a yaml file for models config

### PR DESCRIPTION
**Pull request by maudetes**
Solving #9.
Change the models_config to a yaml format. 
Use libyaml-cpp-dev to do so.

### CODE REVIEW

File: src/rest_server.cpp [hunk 1]
```diff
@@ -12,23 +12,14 @@ rest_server::rest_server(Address addr, std::string &classif_config, int debug) {
...
-      int online = 0;
-      if ((int)vecline.size() > 2) {
-        online = atoi(vecline[2].c_str());
-      }
-      cerr << domain << "\t" << file << "\t" << online << endl;
-      _list_classifs.push_back(new classifier(file, domain));
-    }
+  YAML::Node config = YAML::LoadFile(classif_config);
+
+  for (YAML::const_iterator it=config.begin();it!=config.end();it++){
```
Comment (NoelM) [inline src/rest_server.cpp]: Could be simplified by using :
```c++
for (auto& line : config) {
    string domain = line.first.as<std::string>();
    string file = line.second.as<std::string>();
   _list_classifs.push_back(new classifier(file, domain));
}
```

File: src/rest_server.cpp [hunk 2]
```diff
@@ -12,23 +12,14 @@ rest_server::rest_server(Address addr, std::string &classif_config, int debug) {
...
+    string domain = it->first.as<std::string>();
+    string file = it->second.as<std::string>();
+    cerr << domain << "\t" << file << "\t" << endl;
```
Comment (NoelM) [inline src/rest_server.cpp]: Why do you ask `cerr`?

File: src/rest_server.cpp [hunk 3]
```diff
@@ -12,23 +12,14 @@ rest_server::rest_server(Address addr, std::string &classif_config, int debug) {
...
+    _list_classifs.push_back(new classifier(file, domain));
```
Comment (NoelM) [inline src/rest_server.cpp]: No tests to ensure that settings are good?

**Review by NoelM — COMMENTED** [review]
Things to modify

File: src/rest_server.cpp (hunk 1 above)
Comment (maudetes) [reply src/rest_server.cpp]: Indeed!

File: src/rest_server.cpp (hunk 2 above)
Comment (maudetes) [reply src/rest_server.cpp]: For now, all logs are written to `cerr`.

File: src/rest_server.cpp (hunk 3 above)
Comment (maudetes) [reply src/rest_server.cpp]: Indeed. No test is made to ensure that the yaml file exists. Should we raise an error ?

File: src/rest_server.cpp (hunk 2 above)
Comment (NoelM) [reply src/rest_server.cpp]: This has to be changed, `cout` for logs, and `cerr` reserved for warnings & errors

File: src/rest_server.cpp (hunk 3 above)
Comment (NoelM) [reply src/rest_server.cpp]: First, you've to verify that file exists. Second, verify that fields `first` and `second` are not empty.

**Review by NoelM — COMMENTED** [review]
Answers to comments

File: src/rest_server.cpp (hunk 3 above)
Comment (maudetes) [reply src/rest_server.cpp]: Done

File: src/rest_server.cpp (hunk 2 above)
Comment (maudetes) [reply src/rest_server.cpp]: Done

File: src/rest_server.cpp (hunk 1 above)
Comment (maudetes) [reply src/rest_server.cpp]: Done

**Review by NoelM — APPROVED** [review]
LGTM

**Pull request merged**